spec/classes/foreman_spec.rb

```diff
@@ -42,4 +42,98 @@ class { 'foreman::providers':
       end
     end
   end
+
+  context 'with all parameters' do
+    let :facts do
+      default_facts.merge({
+        :operatingsystem        => 'RedHat',
```

Comment:
If you place this in the loop above, OS facts will be set, rather than only test one OS.